grails-app/services/org/pih/warehouse/data/ProductAssociationDataService.groovy

```diff
@@ -28,8 +29,32 @@ class ProductAssociationDataService {
             productAssociationInstance.product = Product.findByProductCode(params['product.productCode'])
             productAssociationInstance.associatedProduct = Product.findByProductCode(params['associatedProduct.productCode'])
             params['product.id'] =  productAssociationInstance.product?.id
+            params['product.name'] =  productAssociationInstance.product?.name
             params['associatedProduct.id'] =  productAssociationInstance.associatedProduct?.id
+            params['associatedProduct.name'] =  productAssociationInstance.associatedProduct?.name
 
+            params.hasMutualAssociation = Boolean.valueOf(params.hasMutualAssociation as String) || (params.hasMutualAssociation as String)?.equalsIgnoreCase("yes")
+
+            // check for duplicate association entries in the file
+            def indexOfDuplicate = listOfValidatedProductAssociationParams.findIndexOf{it ->
+                if (params.hasMutualAssociation || it.hasMutualAssociation) {
+                    return  (
+                            params['product.id'] == it['product.id'] &&
+                            params['associatedProduct.id'] == it['associatedProduct.id'] &&
+                            params['code'] == params['code']
```

Comment:
Seems like this (`params['code'] == params['code']`) will be always true

Reply:
yep this is a typo